### PR TITLE
refactor: strategies

### PR DESCRIPTION
**Pull request by gregdhill**
- Restrict `GatewayQuoteParams` to be more like `TransferParams`
- Adopt `Token` type from superchain repo
- Add `getStrategies` with Swing types

Process to integrate e.g. Pell Network without an output token:

```ts
const strategies = await gatewaySDK.getStrategies();
const contract = strategies.filter(strategy => strategy.integration.type === "staking")[0];

const params: GatewayQuoteParams = {
  toToken: contract.address, // NOTE: this is actually the strategy contract not the output token
  ...
};

const quote = await gatewaySDK.getQuote(params);
```

### CODE REVIEW

File: sdk/test/gateway.test.ts
Comment (derrekcoleman) [inline sdk/test/gateway.test.ts:1]: Related to my Bitcoin address question above, I suspect the DX (i.e. descriptive errors) around multiple Bitcoin address types will be important here.

Most (all?) of the server-side testing [uses the same mock Bitcoin address](https://github.com/bob-collective/bob-gateway/blob/008d0acb8849992ac24905148a00627d38355e4b/app/src/server.rs#L1211). 

```rs
   fn mock_bitcoin_address() -> bitcoin::Address {
        bitcoin::Address::from_str("bcrt1qj4ymqevserwz045556l6qjycxnumcwwk09t3af")
            .unwrap()
            .assume_checked()
    }
```

I'm happy to help with test coverage (and perhaps the errors, if needed) if you can help me catch up to how we currently handle Bitcoin address validation.

**Review by derrekcoleman — COMMENTED** [review]
My comments are mostly about the DX and UX for Bitcoin address validation.

If we _don't_ have any opinions about what Bitcoin addresses are valid (e.g. p2tr), then I suppose we can ignore most of what I said and leave it up to the UI's implementation. For example, [here's where our UI](https://github.com/bob-collective/ui/blob/c1160742f603e0080b6b91f378c37282de1dbf9e/apps/evm/src/pages/Bridge/components/BridgeForm/BtcBridgeForm.tsx#L366) uses sats-wagmi to give the user a descriptive error for a taproot address.

Here's another example of [stakingrewards.com](https://www.stakingrewards.com/stake-app?input=bitcoin) just throwing up a box to make the user verify they don't hold ordinals in the address they use.
Screenshot 2024-08-29 at 2 14 41 PM

**Review by derrekcoleman — APPROVED** [review]
Also approving so you can merge

File: sdk/src/gateway/types.ts
Comment (gregdhill) [reply sdk/src/gateway/types.ts:6]: Do you have a suggestion for a well-known chain slug? Happy to change it but couldn't think of anything

File: sdk/src/gateway/types.ts
Comment (gregdhill) [reply sdk/src/gateway/types.ts:63]: Only required if the user is using P2SH-P2WPKH, we have some descriptive errors but agree we should note in the docs

**Pull request merged**
